test/e2e/fail_forward_e2e_test.go

```
@@ -0,0 +1,309 @@
+			Expect(magicCatalog.DeployCatalog(context.Background())).To(BeNil())
+
+			By("creating the testing subscription")
+			subscription = &operatorsv1alpha1.Subscription{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      fmt.Sprintf("%s-sub", catalogSourceName),
+					Namespace: ns.GetName(),
+				},
+				Spec: &operatorsv1alpha1.SubscriptionSpec{
+					CatalogSource:          catalogSourceName,
+					CatalogSourceNamespace: ns.GetName(),
+					Channel:                "stable",
+					Package:                "packageA",
+				},
+			}
+			Expect(c.Create(context.Background(), subscription)).To(BeNil())
+
+			By("waiting until the subscription has an IP reference")
+			subscription, err := fetchSubscription(crclient, subscription.GetNamespace(), subscription.GetName(), subscriptionHasInstallPlanChecker)
+			Expect(err).Should(BeNil())
+
+			originalInstallPlanRef := subscription.Status.InstallPlanRef
+
+			By("waiting for the v0.1.0 CSV to report a succeeded phase")
+			_, err = fetchCSV(crclient, subscription.Status.CurrentCSV, ns.GetName(), buildCSVConditionChecker(operatorsv1alpha1.CSVPhaseSucceeded))
+			Expect(err).ShouldNot(HaveOccurred())
+
+			By("updating the catalog with a broken v0.2.0 bundle image")
+			brokenProvider, err := NewFileBasedFiledBasedCatalogProvider(filepath.Join(testdataDir, failForwardTestDataBaseDir, "example-operator.v0.2.0.yaml"))
+			Expect(err).To(BeNil())
+
+			err = magicCatalog.UpdateCatalog(context.Background(), brokenProvider)
+			Expect(err).To(BeNil())
+
+			By("verifying the subscription is referencing a new installplan")
+			subscription, err = fetchSubscription(crclient, subscription.GetNamespace(), subscription.GetName(), subscriptionHasInstallPlanDifferentChecker(originalInstallPlanRef.Name))
+			Expect(err).Should(BeNil())
+
+			By("patching the installplan to reduce the bundle unpacking timeout")
+			addBundleUnpackIPAnnotation(context.Background(), c, objectRefToNamespacedName(subscription.Status.InstallPlanRef), "1s")
```

Should we call this `updateBundleUnpackTimeout` instead? or `addBundleUnpackTimeoutIPAnnotation`?

---

Is this to force an error on the IP? Could it be flaky, or will it never take less than 1s?

---

Is this to force an error on the IP? Could it be flaky, or will it never take less than 1s?

---

The bundle it's trying to unpack here doesn't exist, so there's no flake. This is just to force the unpack job to fail faster, the default timeout was 5 minutes.